### PR TITLE
EUI-4230 Specific Access Request Duration

### PR DESCRIPTION
**Pull request by scott-turner-hmcts**
### JIRA link ###

https://tools.hmcts.net/jira/browse/EUI-4230

### Change description ###

- UI work for **Specific Access Duration** stage
- Created `DurationHelperService` for common functionality.
- Created units test for `DurationHelperService` and `SpecificAccessDurationComponent`

**NOTE** The `ChooseDurationComponent` could be refactored in future to make use of `DurationHelperService` to avoid duplicate functionality for validation, etc.

e.g. this file `role-access/containers/allocate-role/choose-duration/choose-duration.component.ts`

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: src/role-access/containers/specific-access/specific-access-duration/specific-access-duration.component.ts [hunk 1]
```diff
@@ -1,60 +1,197 @@
...
+import { select, Store } from '@ngrx/store';
+import {
+  DurationTypeDescription,
+  Period,
+  SpecificAccessStateData,
+  SpecificAccessNavigation,
+  SpecificAccessNavigationEvent
+} from '../../../models';
+import { DurationType } from '../../../models/enums';
+import { DurationHelperService } from '../../../services/duration-helper.service';
```
Comment (connorpgpmcelroy) [inline src/role-access/containers/specific-access/specific-access-duration/specific-access-duration.component.ts:12]: Minor comment - Getting service from direct file, could it not be put into respective index.ts?

**Review by connorpgpmcelroy — CHANGES_REQUESTED** [review]
Few minor comments

File: src/role-access/containers/specific-access/specific-access-duration/specific-access-duration.component.ts (hunk 1 above)
Comment (scott-turner-hmcts) [reply src/role-access/containers/specific-access/specific-access-duration/specific-access-duration.component.ts]: Have updated import

File: src/role-access/containers/specific-access/specific-access-duration/specific-access-duration.component.spec.ts [hunk 2]
```diff
@@ -1,24 +1,26 @@
-import { CUSTOM_ELEMENTS_SCHEMA } from '@angular/core';
+import { FormBuilder, FormsModule, ReactiveFormsModule } from '@angular/forms';
 import { async, ComponentFixture, TestBed } from '@angular/core/testing';
-import { Store } from '@ngrx/store';
-import { ChooseRadioOptionComponent } from '../../../components';
+import { StoreModule } from '@ngrx/store';
 import { SpecificAccessDurationComponent } from './specific-access-duration.component';
+import { DurationHelperService } from '../../../services/duration-helper.service';
+import { DurationType } from '../../../models/enums';
```
[the remark this answers is not in the document]
Comment (scott-turner-hmcts) [reply src/role-access/containers/specific-access/specific-access-duration/specific-access-duration.component.spec.ts:5]: Have reordered so model is above service import

**Review by connorpgpmcelroy — APPROVED** [review]
Minor comments resolved

**Review by tarikdogan-hmcts — APPROVED** [review]

**Pull request merged**